clients/drcachesim/simulator/cache_fifo.cpp

```diff
@@ -56,8 +56,8 @@ cache_fifo_t::init(int associativity, int block_size, int total_size,
 
     // Create a replacement pointer for each set, and
     // initialize it to point to the first block.
-    for (int i = 0; i < blocks_per_set_; i++) {
-        get_caching_device_block(i << assoc_bits_, 0).counter_ = 1;
+    for (int i = 0; i < blocks_per_way_; i++) {
+        get_caching_device_block_scaled(i, 0).counter_ = 1;
```

Comment:
I'm thinking whether we really need a helper method here; maybe just inline the 1-line implementation of get_caching_device_block_scaled?